File: 100-Days-Series/Day 33/SelectionSort.py

# Function for selection sort
def getSelectionSort(arr):
    for i in range(0,len(arr)-1,1):
        minIndex=i
        for j in range(i+1,len(arr),1):
            if arr[j]<arr[minIndex]:
                minIndex=j
        temp=arr[i]
        arr[i]=arr[minIndex]
        arr[minIndex]=temp
    return arr
        
# Let's Start here
arr=[3,4,2,5,6,8,7,9,1]
print("Before Sorting..")
print(arr)
print("After Selection Sorting..")
print(getSelectionSort(arr))

File: 100-Days-Series/Day 33/SelectionSort.js

// Arrow Function for selection sort
const getSelectionSort=(arr)=>{
    for(let i=0;i<arr.length;i++){
        let minIndex=i;
        for(let j=i+1;j<arr.length;j++){
            if(arr[j]<arr[minIndex])
                minIndex=j;
        }
        let temp=arr[minIndex];
        arr[minIndex]=arr[i];
        arr[i]=temp;
    }
    return arr;
}
// Let's Start here
arr=new Array(4,3,5,6,2,1,7,9,8);
console.log("Before Sorting..");
console.log(arr);
console.log("After Selection Sorting..");
console.log(getSelectionSort(arr));